modules/executive/components/SpellEffectsTab.tsx

```diff
@@ -39,17 +43,168 @@ export function SpellEffectsTab({
+        >
+          <Box
+            sx={{
+              display: 'flex',
+              justifyContent: 'space-between'
+            }}
+          >
+            {spellData?.executiveHash ? (
+              <Text sx={{ mr: 2, fontWeight: 'semiBold', wordBreak: 'break-all' }}>
+                {spellData?.executiveHash}
+              </Text>
+            ) : (
+              <SkeletonThemed width="300px" height="15px" />
+            )}
+
+            <Box sx={{ cursor: 'pointer', ml: 2, minWidth: '99px' }} onClick={() => setExpanded(!expanded)}>
+              <Text color={'textMuted'}>
+                What&apos;s this? <DaiUIIcon name={expanded ? 'chevron_up' : 'chevron_down'} size={2} />
+              </Text>
+            </Box>
+          </Box>
+
+          {expanded && (
+            <Box sx={{ mt: 3 }}>
+              <Text as="p" color="textMuted">
+                This hash allows for manually verifying that the Spell belongs to the correct Executive
+                Proposal. It can be reproduced by hashing the raw markdown text of this Executive Proposal.
+                The hash, the URL to the raw markdown text, and the correct hashing algorithm are all
+                registered on the blockchain in the Spell smart contract.
+              </Text>
+              <Box sx={{ mt: 3 }}>
+                <ThemeUILink
+                  href={'https://makerdao.world/en/learn/governance/audit-exec-spells'}
+                  target="_blank"
+                >
+                  <Text sx={{ color: 'accentBlue', ':hover': { color: 'blueLinkHover' } }}>
+                    Learn more about auditing Executive Spells
+                    <DaiUIIcon ml={2} name="arrowTopRight" size="2" />
+                  </Text>
+                </ThemeUILink>
+              </Box>
+            </Box>
+          )}
+        </Box>
+      </Box>
+      <Box sx={{ display: 'flex', flexWrap: 'wrap' }}>
+        {'proposalLink' in proposal && <Box sx={{ width: ['100%', '50%'] }}>
+          <Text
+            as="p"
+            sx={{
+              fontWeight: 'semiBold'
+            }}
+          >
+            Executive Hash
           </Text>
-        </ThemeUILink>
-      </Flex>
+          <Box>
+            <ThemeUILink href={proposal.proposalLink} target="_blank">
+              <Text sx={{ color: 'accentBlue', ':hover': { color: 'blueLinkHover' } }}>
+                View in GitHub
+                <DaiUIIcon ml={2} name="arrowTopRight" size="2" />
+              </Text>
+            </ThemeUILink>
+          </Box>
+        </Box>}
+        <Box sx={{ width: ['100%', '50%'] }}>
+          <Box mb={3} mt={[3, 0]} sx={{ display: 'flex' }}>
+            <Box
+              mr={2}
+              sx={{
+                color: 'primary',
+                background: '#F6F8F9',
+                borderRadius: '100%',
+                width: '34px',
+                minWidth: '34px',
+                height: '34px',
+                display: 'flex',
```

Comment:
Just FYI you can use `<Flex/>` component instead of `<Box/>` when it's `display: flex`